recipe/meta.yaml

```
@@ -43,6 +43,7 @@ requirements:
     - scipy
     - libxcrypt               # [linux and py<39]
     {% if cuda_major >= 12 %}
+    - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
```

This whole block is already guarded, no need to have a selector 
```suggestion
    - cuda-version {{ cuda_compiler_version }}
```